README.md

```diff
@@ -1,93 +1,142 @@
+
+### `requestReset()` method
+
+This re-renders all of the currently-displayed items, updating them from their source data using `updateChild`.
+
+This can be useful for if you mutate the `items` array, or elements in it, instead of setting the `items` property to a new value. Also see [the example below](#data-manipulation).
+
+_We are discussing the naming of this API, as well as whether it should exist at all, in [#26](https://github.com/PolymerLabs/virtual-list/issues/26). The aforementioned [#29](https://github.com/PolymerLabs/virtual-list/issues/29) is also relevant._
+
+### "`rangechange`" event
+
+Bubbles: false / Cancelable: false / Composed: false
+
+Fired when the list has finished rendering a new range of items, e.g. because the user scrolled. The event is an instance of `RangeChangeEvent`, which has the following properties:
+
+- `first`: a number giving the index of the first item currently rendered.
+- `last`: a number giving the index of the last item currently rendered.
+
+Also see [the example below](#range-changes).
+
+## More examples
+
+### DOM recycling
 
 You can recycle DOM by using the `recycleChild` function to collect DOM, and reuse it in `newChild`.
 
-Ensure to perform DOM updates in `updateChild`, as recycled children will still be configured with the old `item`.
+When doing this, be sure to perform DOM updates in `updateChild`, as recycled children will otherwise have the data from the previous `item`.
 
 ```js
-const recycled = [];
+const nodePool = [];
 
 Object.assign(list, {
-  newChild: (item, index) => {
-    return recycled.pop() || document.createElement('section');
+  newChild() {
+    return nodePool.pop() || document.createElement('section');
   },
-  updateChild: (child, item, index) => {
+  updateChild(child, item, index) {
     child.textContent = index + ' - ' + item.name;
   },
-  recycleChild: (child, item, index) => {
-    recycled.push(child);
+  recycleChild(child) {
+    nodePool.push(child);
   }
 };
 ```
 
-## Data manipulation
+### Data manipulation
+
+The `<virtual-list>` element will automatically rerender the displayed items when it receives a new `items` array. For example, to add a new item to the end, you could do:
 
-`<virtual-list>` will rerender the displayed items when it receives a new `items` array.
 ```js
 list.items = list.items.concat([{name: 'new item'}]);
 ```
 
-If you want to keep the same `items` array instance, ensure to update the DOM via `updateChild` and to use `requestReset()` to rerender the displayed items.
+In this case, `newChild` will be called for every item, including the ones that already had corresponding elements in the old items array.
+
+If you want to keep the same `items` array instance, you can use `requestReset()` to rerender the currently-displayed items. If you do this, you'll also need to set `updateChild`, since the elements will no longer be created from scratch. For example:

```

Comment:
Probably here we can be more explicit on when to use `updateChild`: the user should always set `updateChild` if they plan to update the array contents, specifically if the dom was already created for that index. e.g.:
```js
list.items = ['item'];
requestAnimationFrame(() => {
  // The list has already created a child for index = 0
  // so the user needs to define updateChild in order 
  // to update that DOM node.
  // The new item (index = 1) will be created via newChild
  list.items = ['changed!', 'new item'];
});
```
In short, it's just safer to update the DOM with `updateChild` if you plan to manipulate data.
Maybe it makes sense to mention this at the top of this section?